File: Assignment3-master/Assignment3-master/Java/Collatz.java
import java.util.*;

public class Collatz
{
    public static void main(String[] args)
    {
        long number = 0;
        int finish = 0;
        int numCount = 0;
        int numCountCopy = 0;
        int longestSequence = 0;
        System.out.print("Enter the end of range integer of sequence (Eg. if input 100 = sequence ends at 100) : ");
        Scanner scanner = new Scanner(System.in);
        finish = scanner.nextInt();
    
        for (int i = 2; i <= finish; i++) //number should not start under 1
        {   
            number = i; //resets number to next value that needs to be tested
            numCount = 0; //resets sequence count to 0 every time
    
            while (number != 1) //keep going til number hits 1
            {   
                if (EvenOrNot(number) == 1) //evaluates if number is even or odd, if it is do the even calculations
                {
                    number = number / 2;
                    numCount++;
                }
                else //if its odd, do the odd calculations
                {
                    number = (number * 3) + 1;
                    numCount++;
                 }
            }
        
            if (numCount > numCountCopy) //if the prev sequence count is bigger then current keep that value, the biggest value gets left in longestSequence
            {
                longestSequence = i;
                numCountCopy = numCount;
            }
        }
    
        System.out.println("Number with biggest sequence is : " + longestSequence);
    }

    public static int EvenOrNot (long number)
    {
        if ( number % 2 == 0)
            return 1;
        else
            return 0;
    }

}

File: Assignment3-master/Assignment3-master/Fortran/Collatz.f90
program Collatz
    integer :: num
    integer :: finish
    integer :: numCount
    integer :: numCountCopy
    integer :: longestSequence

    print *, "Enter the end of range integer of sequence (Eg. if input 100 = sequence ends at 100) : "
    read(*,*) finish
    do i = 2, finish, 1 !ONLY WORKS IF THERE IS A PRINT STATEMENT INSIDE LOOP
        num = i
        numCount = 0
        do while (num .ne. 1)
            if (MODULO(num, 2) .eq. 0) then !if even
                num = num / 2
                numCount = numCount + 1
            else if (MODULO(num, 2) .ne. 0) then !if odd
                num = (num * 3) + 1
                numCount = numCount + 1
            end if
        end do
        if (numCount .gt. numCountCopy) then
            longestSequence = i
            numCountCopy = numCount
        end if
        print *, num !PRINT STATEMENT NEEDED TO MAKE IT WORK
    end do

    print *, "Number with biggest sequence is : ", longestSequence
end program Collatz
